py/dml/c_backend.py

```diff
@@ -3291,6 +3320,8 @@ def generate_cfile_body(device, footers, full_module, filename_prefix):
     generate_events(device)
     generate_identity_data_decls()
     generate_object_vtables_array()
+    if compat.shared_logs_on_device not in dml.globals.enabled_compat:
+        generate_log_object_assoc_array()
```

Comment:
naming: this is fine, but makes me relate to the unrelated term "associative array", often used for the dict datatype (in particular in the scheme/lisp world)

Reply:
Ah, true, didn't consider that... Intended meaning was "log object assocations." Will think over if I want to change it.

Reply:
`generate_log_object_assocs_array`, ez